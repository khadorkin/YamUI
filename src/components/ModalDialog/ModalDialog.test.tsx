/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { ModalDialog, ModalDialogProps } from './index';
import { ModalDialogSize } from './types';
import { KeyCodes } from '../../util/keyCodes';

describe('<ModalDialog />', () => {
  let component: ShallowWrapper<ModalDialogProps>;
  let onDismissed: jest.Mock;

  describe('with default options', () => {
    beforeEach(() => {
      onDismissed = jest.fn();
      component = shallow(
        <ModalDialog
          title={'Title'}
          isOpen={false}
          onDismissed={onDismissed}
          size={ModalDialogSize.MEDIUM}
          closeAriaLabel={'close'}
        />,
      );
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with additional className', () => {
    beforeEach(() => {
      onDismissed = jest.fn();
      component = shallow(
        <ModalDialog
          title={'Title'}
          className="TEST_CLASSNAME"
          isOpen={false}
          onDismissed={onDismissed}
          size={ModalDialogSize.MEDIUM}
          closeAriaLabel={'close'}
        />,
      );
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('when the title is hidden', () => {
    beforeEach(() => {
      onDismissed = jest.fn();
      component = shallow(
        <ModalDialog
          title={'Title'}
          className="TEST_CLASSNAME"
          isOpen={false}
          onDismissed={onDismissed}
          size={ModalDialogSize.MEDIUM}
          closeAriaLabel={'close'}
          hideTitle={true}
        />,
      );
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('when the modal is open', () => {
    beforeEach(() => {
      onDismissed = jest.fn();
      component = shallow(
        <ModalDialog
          title={'Title'}
          className="TEST_CLASSNAME"
          isOpen={true}
          onDismissed={onDismissed}
          size={ModalDialogSize.MEDIUM}
          closeAriaLabel={'close'}
          hideTitle={true}
        />,
      );
    });

    describe('when the escape key is pressed', () => {
      beforeEach(() => {
        const event = new KeyboardEvent('keydown', { keyCode: KeyCodes.escape } as any);
        document.dispatchEvent(event);
        component.update();
      });

      it('closes the modal', () => {
        expect(component.props().isOpen).toBe(false);
      });
    });

    describe('when any other key is pressed', () => {
      beforeEach(() => {
        const event = new KeyboardEvent('keydown', { keyCode: KeyCodes.space } as any);
        document.dispatchEvent(event);
        component.update();
      });

      it('keeps the modal open', () => {
        expect(component.props().isOpen).toBe(true);
      });
    });
  });
});
