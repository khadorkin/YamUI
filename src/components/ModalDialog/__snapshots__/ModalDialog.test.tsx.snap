// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalDialog /> when the title is hidden matches its snapshot 1`] = `
<WithResponsiveMode
  className=""
  containerClassName=
      y-modalDialog
      TEST_CLASSNAME
      {
        width: 580px;
      }
  isBlocking={false}
  isDarkOverlay={true}
  isOpen={false}
  onDismiss={[MockFunction]}
  onDismissed={[MockFunction]}
>
  <Block
    horizontalPadding="xLarge"
    verticalPadding="medium"
  >
    <ScreenreaderText>
      <h1>
        Title
      </h1>
    </ScreenreaderText>
  </Block>
</WithResponsiveMode>
`;

exports[`<ModalDialog /> with additional className matches its snapshot 1`] = `
<WithResponsiveMode
  className=""
  containerClassName=
      y-modalDialog
      TEST_CLASSNAME
      {
        width: 580px;
      }
  isBlocking={false}
  isDarkOverlay={true}
  isOpen={false}
  onDismiss={[MockFunction]}
  onDismissed={[MockFunction]}
>
  <Block
    horizontalPadding="xLarge"
    verticalPadding="medium"
  >
    <FixedGridRow
      bottomSpacing="xSmall"
    >
      <FixedGridColumn>
        <Heading
          level="1"
          size="5"
        >
          Title
        </Heading>
      </FixedGridColumn>
      <FixedGridColumn
        fixed={true}
      >
        <Clickable
          ariaLabel="close"
          block={true}
          onClick={[MockFunction]}
          unstyled={true}
        >
          <Cancel
            block={true}
            size="12"
          />
        </Clickable>
      </FixedGridColumn>
    </FixedGridRow>
  </Block>
</WithResponsiveMode>
`;

exports[`<ModalDialog /> with default options matches its snapshot 1`] = `
<WithResponsiveMode
  className=""
  containerClassName=
      y-modalDialog
      {
        width: 580px;
      }
  isBlocking={false}
  isDarkOverlay={true}
  isOpen={false}
  onDismiss={[MockFunction]}
  onDismissed={[MockFunction]}
>
  <Block
    horizontalPadding="xLarge"
    verticalPadding="medium"
  >
    <FixedGridRow
      bottomSpacing="xSmall"
    >
      <FixedGridColumn>
        <Heading
          level="1"
          size="5"
        >
          Title
        </Heading>
      </FixedGridColumn>
      <FixedGridColumn
        fixed={true}
      >
        <Clickable
          ariaLabel="close"
          block={true}
          onClick={[MockFunction]}
          unstyled={true}
        >
          <Cancel
            block={true}
            size="12"
          />
        </Clickable>
      </FixedGridColumn>
    </FixedGridRow>
  </Block>
</WithResponsiveMode>
`;
